File: src/main/java/io/github/takenoha/towerdefense/config/TerrainMutationSettings.java
package io.github.takenoha.towerdefense.config;

/**
 * Operator-controlled evidence flags for the experimental terrain-mutation path.
 *
 * <p>All flags default to {@code false}. They are deliberately kept separate so that an
 * operator cannot accidentally treat a Paper integration check as proof that recovery is safe.
 */
public record TerrainMutationSettings(
        boolean requested,
        boolean paperIntegrationVerified,
        boolean recoveryVerified) {

    /** Returns the fail-closed default used by older configurations and direct constructors. */
    public static TerrainMutationSettings disabled() {
        return new TerrainMutationSettings(false, false, false);
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/config/TerrainMutationSettings.kt
package io.github.takenoha.towerdefense.config

/** Operator-controlled evidence flags for the experimental terrain-mutation path. */
@JvmRecord
data class TerrainMutationSettings(
    val requested: Boolean,
    val paperIntegrationVerified: Boolean,
    val recoveryVerified: Boolean,
) {
    companion object {
        /** Returns the fail-closed default used by older configurations and direct constructors. */
        @JvmStatic
        fun disabled(): TerrainMutationSettings = TerrainMutationSettings(false, false, false)
    }
}
